### ISSUE TITLE
Input: has incorrect typings for `onChange`

### ISSUE THREAD

**Issue by tomitrescak** [Semantic-Org/Semantic-UI-React#3229]
Hi, 

0.82.5 comes with a major breaking change that makes all of our typescript codebases error out on event handlers. Previously you had event handler on form elements correctly typed, now there appears only a SyntheticEvent.

For example, this:

```ts
export interface InputProps { 
...
onChange?: (event: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData) => void
}
```

changed to this:

```ts
export interface InputProps { 
...
onChange?: (event: React.SyntheticEvent<HTMLInputElement>, data: InputOnChangeData) => void
}
```

Is this a bug or a feature? Shall we start changing our code bases? I hope we can bring the old functionality back as getting type safe handlers just become very difficult.

**Comment by tomitrescak** [comment]
Nevermind .. it was probably the new react typings that became more strict to break this. Your typings have not changed.

**Comment by tomitrescak** [comment]
It would be nice to have correct handlers though ... reopening.

**Comment by layershifter** [comment]
Hm, as I see there was _always_ `React.SyntheticEvent` event:

https://github.com/Semantic-Org/Semantic-UI-React/blob/6a3e447c29aa72c75ff217ab08455e039e4c6c79/src/elements/Input/Input.d.ts#L61

However, these changes makes a sence because `ChangeEvent` contains `target`:
```tsx
    interface ChangeEvent<T = Element> extends SyntheticEvent<T> {
        target: EventTarget & T;
    }
```

**Comment by layershifter** [comment]
Fixed in #3230.

**Comment by tomitrescak** [comment]
Yah I noticed that the typing did not change. But thanks for the fix! Amazing
